crates/subspace-networking/src/node_runner.rs

```diff
@@ -657,7 +658,39 @@ where
 
                 kademlia.remove_peer(&peer_id);
             }
+
+            self.add_observed_external_address(info.observed_addr);
```

Comment:
Should we only add observed non-private addresses here maybe?

Reply:
It's possible to add this filter with `--enable-private-ips` dependency. However, it shouldn't be necessary because it's either private because it's a local setup or public because it was dialed successfully to the listening port.

Reply:
Also looks a bit like potential attack vector, I think we might want to skip this entirely if user provided external address explicitly.

Reply:
Yes, it is. This operation should be enabled in the trusted enviroment as suggested by libp2p: https://github.com/libp2p/rust-libp2p/blob/master/protocols/identify/CHANGELOG.md#0430

Reply:
Also does it include identity at the end? I'm not sure if something in libp2p verifies it, but we probably want to if not.

Reply:
The libp2p inner code chains external addresses and listening addresses (which are without identity) in the same vector when filling the provider addresses vector. So, it should work without identity in the end.